index.js

```diff
@@ -1,20 +1,748 @@
-const readline = require('readline');
-const readlineInterface = readline.createInterface(process.stdin, process.stdout);
+const readline = require("readline");
+const readlineInterface = readline.createInterface(
+  process.stdin,
+  process.stdout
+);
 
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     readlineInterface.question(questionText, resolve);
   });
 }
 
+//Main class for our rooms that serves as a generalized version that can adapt to various instances
+class Room {
+  //our main variables for the constructor are intro: initial room description,
+  //secretItem: changes room to room but default is a key, items: represent mutable items in the room that can be interacted with i.e. picked up and placed in inventory,
+  //props: immutable items in a room that cannot be interacted with, toolbox: for one specific room we are considering using extends for instead
+  constructor(intro, secretItem, items, props, toolbox) {
+    this.intro = intro;
+    this.secretItem = secretItem || "key";
+    this.items = items;
+    this.props = props;
+    this.toolbox = "closed";
+  }
+  //inside the toolbox is a map and can only be revealed if the toolbox is open
+  revealMap() {
+    if (this.toolbox === "open") {
+      this.secretItem = "map";
+      console.log(`Oh wow there's a map`);
+    }
+  }
+
+  read() {
+    return this.intro;
+  }
+
+  //changes the state of the toolbox when the player interacts with the item
+  openToolbox() {
+    if (this.toolbox === "closed") {
+      //default state is closed
+      this.toolbox = "open"; //when method is called state changes to open and will remain open if player comes back to item after interacting with it
+      console.log(`You just opened the toolbox`);
+    } else if (this.toolbox === "open") {
+      // will remain open
+      console.log(`This tool box has already been open`);
+    }
+  }
+  
+  revealMap() {
+    if (this.toolbox === "open") {
+      this.secretItem = "map";
+      console.log(`Oh wow there's a map`);
+    }
+  }
+
+  //will send items from room inventory to player inventory
+  sendItems() {
+    //if the secret item is a map then the user gets a specific message
+    //the way the game is set up now there is only one secret item in each room so there is no need to isolate the correct secret item
+    //and then pop() it out of an array and send it to player inventory
+    if (this.secretItem === "map") {
+      player.inventory.push(this.secretItem);
+      console.log(`You just put a map in your inventory`); //map message to player
+    }
+    //the default operation for this method is to pop off the current item in the room's array
+    let poppedItem = this.items.pop();
+    player.inventory.push(poppedItem); //send the value of the .pop() to player inventory
+    console.log(`You just added ${this.items} to your inventory`); //message to user of what item they just placed in their inventory
+  }
```

Comment:
This is a good start to inventory management. One hangup as this block of code sits right now, is that the player can continue to "take" items after the first item is grabbed. For instance I could take multiple notebooks and check my players inventory and see multiple "items".